### PR TITLE
[Keyboard] Add new keyboard quick17

### PR DESCRIPTION
**Pull request by MakotoKurauchi**
## Description

Added new keyboard Quick17

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* (None)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/yushakobo/quick17/config.h [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+/*
+ * Feature disable options
+ *  These options are also useful to firmware size reduction.
+ */
+
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
```
Comment (drashna) [inline keyboards/yushakobo/quick17/config.h:171]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/yushakobo/quick17/config.h (hunk 1 above)
Comment (MakotoKurauchi) [reply keyboards/yushakobo/quick17/config.h]: thanks!

**Review by drashna — APPROVED** [review]

File: keyboards/yushakobo/quick17/config.h [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+#define RGB_DI_PIN D2
+#define RGBLED_NUM 18
+
+#ifdef RGB_MATRIX_ENABLE
+#   define DRIVER_LED_TOTAL RGBLED_NUM
+#   define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+// #   define RGB_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
+// #   define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#   define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```
Comment (drashna) [inline keyboards/yushakobo/quick17/config.h:67]:
```suggestion
#   define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/yushakobo/quick17/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "quick17_prefs.h"
```
Comment (drashna) [inline keyboards/yushakobo/quick17/keymaps/default/keymap.c:17]:
```suggestion
```

File: keyboards/yushakobo/quick17/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "quick17_prefs.h"
```
Comment (drashna) [inline keyboards/yushakobo/quick17/keymaps/via/keymap.c:17]:
```suggestion
```

File: keyboards/yushakobo/quick17/quick17_prefs.c [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/yushakobo/quick17/quick17_prefs.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/yushakobo/quick17/rules.mk [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/yushakobo/quick17/rules.mk:21]:
```suggestion
```

File: keyboards/yushakobo/quick17/readme.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# quick17
+
+![quick17](https://i.imgur.com/8SkWq0m.jpeg)
+
+*A simple macropad for the beginners*
+
+* Keyboard Maintainer: [yushakobo](https://github.com/yushakobo)
+* Hardware Supported: *Quick17 PCBs, Pro Micro (ATmega32u4)*
```
Comment (fauxpark) [inline keyboards/yushakobo/quick17/readme.md:8]:
```suggestion
A simple macropad for the beginners

* Keyboard Maintainer: [yushakobo](https://github.com/yushakobo)
* Hardware Supported: Quick17 PCBs, Pro Micro (ATmega32u4)
```

File: keyboards/yushakobo/quick17/info.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{
+    "keyboard_name": "quick17",
+    "url": "https://shop.yushakobo.jp/products/quick17",
+    "maintainer": "yushakobo",
+    "width": 6,
+    "height": 3,
```
Comment (fauxpark) [inline keyboards/yushakobo/quick17/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/yushakobo/quick17/quick17_prefs.c (hunk 5 above)
Comment (MakotoKurauchi) [reply keyboards/yushakobo/quick17/quick17_prefs.c:17]: thanks

File: keyboards/yushakobo/quick17/rules.mk [hunk 9]
```diff
@@ -0,0 +1,28 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/yushakobo/quick17/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/yushakobo/quick17/quick17_prefs.h [hunk 10]
```diff
@@ -0,0 +1,12 @@
+enum layer_names {
```
Comment (drashna) [inline keyboards/yushakobo/quick17/quick17_prefs.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/yushakobo/quick17/quick17_prefs.h (hunk 10 above)
Comment (MakotoKurauchi) [reply keyboards/yushakobo/quick17/quick17_prefs.h:17]: thanks!

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**